src/pocket_note/lib/ui/screens/addpowerbill/store/add_power_bill_store.dart

```diff
@@ -65,28 +65,15 @@ abstract class AddPowerBillBase with Store {
       neighborsTotalValue: neighborsTotalValue,
       neighborsTotalReadingInKWm: neighborsTotalReadingInKWm,
     );
-    uiState = AddPowerBillUIState(
-      resultType: ResultType.loading,
-    );
-    savePowerBillUseCase
-        .invoke(powerBill)
-        .then((value) => _onSaveSuccess(value))
-        .catchError((stackTrace) => _onError);
-  }
+    uiState = AddPowerBillUIState(isLoading: true);
 
-  _onSaveSuccess(bool value) {
-    if (value) {
-      uiState = AddPowerBillUIState(
-        resultType: ResultType.success,
-      );
-    } else {
-      _onError();
+    try {
+      bool result = await savePowerBillUseCase.invoke(powerBill);
+      uiState = AddPowerBillUIState(isLoading: false);
```

Comment:
The `uiState` copulation couldn't stay at finally block of try catch?

Reply:
It is an incredible suggestion. I am doing right now!! 💯